src/js/actions/superselect.js

```diff
@@ -670,19 +665,20 @@ define(function (require, exports) {
                             };
 
                             return adapterOS.postEvent(dragEvent);
-                        }
-                    })
-                    .catch(function () {}) // Move should silently fail if there are no selected layers
-                    .finally(function () {
-                        // Re show the transform controls, which will appear once we're out of the modal state
-                        return descriptor.playObject(
-                            toolLib.setToolOptions("moveTool", { "$Abbx": true }),
-                            { canExecuteWhileModal: true });
-                    });
+                        })
+                        .then(function () {
+                            return descriptor.playObject(
+                                toolLib.setToolOptions("moveTool", { "$Abbx": true }),
+                                { canExecuteWhileModal: true }
+                            );
+                        });
+                } else {
+                    return this.dispatchAsync(events.panel.SUPERSELECT_MARQUEE, { x: x, y: y, enabled: true });
+                }
             });
     };
     drag.action = {
-        reads: [locks.JS_DOC, locks.JS_UI],
+        reads: [locks.JS_DOC],
```

Comment:
Ugh, sorry, I may have lead you astray last time, but it seems like this _does_ still read the UI store. See above `uiStore.transformWindowToCanvas(x, y)`.